lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb

```diff
@@ -236,45 +259,53 @@ def self.upload_binary(app_id, binary_path)
           request.headers["Authorization"] = "Bearer " + auth_token
         end
       rescue Faraday::ResourceNotFound
-        UI.crash!(ErrorMessage::INVALID_APP_ID)
+        UI.crash!("#{ErrorMessage::INVALID_APP_ID}: #{app_id}")
       rescue Errno::ENOENT
-        UI.crash!(ErrorMessage::APK_NOT_FOUND)
+        UI.crash!("#{ErrorMessage::APK_NOT_FOUND}: #{binary_path}")
       end
 
+      # Uploads the binary
+      #
+      # Returns the id of the release. Only happens on a successful release, on a fail it crashes.
       def self.upload(app_id, binary_path)
         upload_token = get_upload_token(app_id, binary_path)
-        status = upload_status(app_id, upload_token)
-        if status == "SUCCESS"
-          UI.message("This APK/IPA has been uploaded before. Skipping upload step.")
+        upload_status_response = get_upload_status(app_id, upload_token)
+        if upload_status_response.success?
+          UI.success("This APK/IPA has been uploaded before. Skipping upload step.")
         else
           UI.message("This APK has not been uploaded before.")
           MAX_POLLING_RETRIES.times do
-            if status == "SUCCESS"
-              UI.message("Uploaded APK/IPA Successfully!")
+            if upload_status_response.success?
+              UI.success("Uploaded APK/IPA Successfully!")
               break
-            elsif status == "IN_PROGRESS"
+            elsif upload_status_response.in_progress?
               sleep(POLLING_INTERVAL_SECONDS)
             else
               UI.message("Uploading the APK/IPA.")
               upload_binary(app_id, binary_path)
             end
-            status = upload_status(app_id, upload_token)
+            upload_status_response = get_upload_status(app_id, upload_token)
           end
-          if status != "SUCCESS"
-            UI.message("It took longer than expected to process your APK/IPA, please try again")
+          unless upload_status_response.success?
+            UI.error("It took longer than expected to process your APK/IPA, please try again.")
+            return nil
           end
         end
+        upload_status_response.release_id
       end
 
-      def self.upload_status(app_id, app_token)
+      # Gets the upload status for the app release
+      #
+      # Returns the status of the release.
```

Comment:
"Returns the status of the release." is saying basically the same thing as "Gets the upload status for the app release" so you can remove it unless you want to elaborate.